test/sqllogictest/comment.slt

```diff
@@ -243,20 +243,38 @@ CREATE TYPE int4_list AS LIST (ELEMENT TYPE = int4);
 statement ok
 COMMENT ON TYPE int4_list IS 'supercool_list';
 
+statement ok
+CREATE TYPE custom_type AS (x integer, y int4_list);
```

Comment:
Need a test with a `CREATE TYPE` that makes a map or list, then asserts `COMMENT ON` returns an error on it.

Reply:
Added test for map and list types